packages/chart/src/components/EditorPanel/EditorPanel.tsx

```diff
@@ -1003,6 +1026,25 @@ const EditorPanel = () => {
       columns: updatedColumns
     })
   }
+
+  const colorCodeByCategory = config.visualizationType === 'Bar' && config.visualizationSubType === 'regular' && config.runtime.seriesKeys.length === 1
+  const getLegendColumns = () => {
+    const colorCodeData = data.map(d => d[config.legend.colorCode])
+    return colorCodeByCategory ? colorCodeData : getColumns(false).filter(d => d !== config.xAxis.dataKey)
+  }
+
+  useEffect(() => {
+    const updatedSeriesHighlight = [...(legendIsolateValues || [])]
+    updateField('legend', null, 'seriesHighlight', updatedSeriesHighlight)
+  }, [legendIsolateValues])
+
+  useEffect(() => {
```

Comment:
is there somewhere these can be called declaratively rather than adding more useEffect?